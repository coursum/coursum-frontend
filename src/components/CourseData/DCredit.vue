<template>
  <span v-if="credit">
    {{ credit }}
  </span>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'DCredit',
  props: {
    idx: {
      type: Number,
      default: 0,
    },
  },

  computed: {
    credit(): number | null | undefined {
      return this.$store.state.courseDatas[this.idx].credit;
    },
    curLang(): string {
      return this.$i18n.locale;
    },
  },

});
</script>
