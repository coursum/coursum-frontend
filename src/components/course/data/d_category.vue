<template>
  <div
    class="px-6 text-truncate"
  >
    {{ categoryData ? categoryData : $t("others") }}
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import { basicTemplate } from '@/assets/CourseInfo';

export default Vue.extend({
  name: 'DCategory',
  props: {
    category: { type: Object, default: basicTemplate },
  },
  computed: {
    categoryData(): string | null | undefined {
      return this.category?.[this.curLang];
    },
    curLang(): string {
      return this.$i18n.locale;
    },
  },
});
</script>

<style scoped>
div{
  font-size: 0.7rem;
  color: gray;
}
</style>

<i18n>
{
  "en": {
    "others": "others"
  },
  "jp": {
    "others": "その他"
  }
}
</i18n>
