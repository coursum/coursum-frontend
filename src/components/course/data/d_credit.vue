<template>
  <v-chip
    v-if="credit"
    style="color: #929292"
    outlined
    x-small
  >
    {{ credit }}{{ $t("credit") }}
  </v-chip>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'DCredit',
  props: {
    credit: {
      type: Number,
      default: undefined,
    },
  },
});
</script>

<i18n>
{
  "en": {
    "credit": "credit"
  },
  "jp": {
    "credit": "単位"
  }
}
</i18n>
