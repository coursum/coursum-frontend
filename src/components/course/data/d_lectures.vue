<template>
  <div
    v-if="lecturers && 0 < lecturers.length"
    :class="{'text-truncate': textTruncate}"
    class="px-6"
  >
    <template
      v-for="(lecturer, i) in lecturers"
    >
      <span
        v-if="lecturer.name && lecturer.name[`${curLang}`]"
        :key="i"
        style="display:inline-block;"
        class="mr-2"
      >
        {{ lecturer.name[`${curLang}`] }}
      </span>
    </template>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';

export default Vue.extend({
  name: 'DLectures',
  props: {
    lecturers: {
      type: Array,
      default() {
        return undefined;
      },
    },
    textTruncate: {
      type: Boolean,
      default: true,
    },
  },
  computed: {
    curLang(): string {
      return this.$i18n.locale;
    },
  },
});
</script>

<style scoped>
span{
  font-size: 0.85rem;
  color: var(--v-wordcolor-base);
}
</style>
