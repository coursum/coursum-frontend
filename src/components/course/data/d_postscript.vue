<template>
  <span
    v-if="postscriptData"
    style="color: #929292"
    class="px-6 text-caption text-truncate"
  >
    {{ postscriptData }}
  </span>
</template>

<script lang="ts">
import { PropType, computed, defineComponent } from '@vue/composition-api';

import { Basic, basicTemplate } from '@/assets/CourseInfo';

export default defineComponent({
  name: 'DPostsctipt',
  props: {
    postscript: {
      type: Object as PropType<Basic>,
      default: basicTemplate,
    },
  },
  setup: (props, context) => {
    const curLang = computed(() => context.root.$i18n.locale as 'jp' | 'en');

    const postscriptData = computed((): string | null | undefined => props
      .postscript?.[curLang.value]);

    return {
      postscriptData,
    };
  },
});
</script>
