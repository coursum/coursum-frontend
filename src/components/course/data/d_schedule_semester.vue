<template>
  <v-chip v-if="semesterData"
          color="caption"
          outlined
          x-small
  >
    {{ semesterData }}
  </v-chip>
</template>

<script lang="ts">
import type { PropType } from '@vue/composition-api';
import { computed, defineComponent } from '@vue/composition-api';
import type { Course } from 'coursum-types';

export default defineComponent({
  name: 'DScheduleSemester',
  props: {
    semester: {
      type: Object as PropType<Course['schedule']['semester']>,
      required: true,
    },
  },
  setup: (props, context) => {
    const curLang = computed(() => context.root.$i18n.locale as 'en' | 'ja');
    const semesterData = computed(() => props.semester[curLang.value]);

    return {
      semesterData,
    };
  },
});
</script>
