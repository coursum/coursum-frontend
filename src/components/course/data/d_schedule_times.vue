<template>
  <span v-if="courseTimes">
    <v-chip v-for="(time, i) in courseTimes" :key="i"
            color="caption"
            outlined
            x-small
    >
      {{ time }}
    </v-chip>
  </span>
</template>

<script lang="ts">
import type { PropType } from '@vue/composition-api';
import { computed, defineComponent } from '@vue/composition-api';
import type { Course } from 'coursum-types';

export default defineComponent({
  name: 'DScheduleTimes',
  props: {
    times: {
      type: Object as PropType<Course['schedule']['times']>,
      required: true,
    },
  },
  setup: (props, context) => {
    const curLang = computed(() => context.root.$i18n.locale as 'en' | 'ja');

    const courseTimes = computed(() => {
      const times = props.times[curLang.value] || '';

      return times.split(',').map((time) => time.trim());
    });

    return {
      courseTimes,
    };
  },
});
</script>
