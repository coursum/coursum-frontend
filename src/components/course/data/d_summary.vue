<template>
  <div
    v-if="summaryData"
    :class="{summary: textTruncate}"
    class="px-6 primary--text py-1"
    :style="summaryStyle"
  >
    <p
      class="my-0"
    >
      {{ summaryData }}
    </p>
  </div>
</template>

<script lang="ts">
import Vue from 'vue';
import { basicTemplate } from '@/assets/CourseInfo';

export default Vue.extend({
  name: 'DSummary',
  props: {
    summary: {
      type: Object,
      default: basicTemplate,
    },
    title: {
      type: Object,
      default: basicTemplate,
    },
    textTruncate: {
      type: Boolean,
      default: true,
    },
  },
  data() {
    return {
      summaryStyle: { 'font-size': '1rem' },
    };
  },
  computed: {
    summaryData(): string | null | undefined {
      if (this.titleJp === '研究会Ａ' || this.titleJp === '研究会Ｂ') {
        return this.summary?.en;
      }

      return this.summary?.[this.curLang];
    },
    titleJp(): string | null | undefined {
      return this.title?.jp;
    },
    curLang(): string {
      return this.$i18n.locale;
    },
  },
});
</script>

<style scoped>
@import url('https://fonts.googleapis.com/css2?family=Noto+Sans+JP&display=swap');

.summary p {
  display: -webkit-box;
  -webkit-box-orient: vertical;
  -webkit-line-clamp: 5;
  overflow: hidden;
}
</style>
