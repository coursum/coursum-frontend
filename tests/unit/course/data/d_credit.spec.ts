import { shallowMount } from '@vue/test-utils';

import { creditMock } from '../../../data';
import { i18n } from '../../../setup';

import DCredit from '@/components/course/data/d_credit.vue';

const correct = creditMock;

const options = {
  i18n,
  propsData: {
    credit: creditMock,
  },
  mocks: {
    $t: (msg: string) => msg,
  },
};

describe('DCredit', () => {
  it('should render correct markup when prop is correct', () => {
    options.propsData.credit = correct;
    const wrapper = shallowMount(DCredit, options);

    const vChip = wrapper.find('v-chip-stub');
    expect(vChip.text()).toEqual(`${correct}credit`);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('should not render anything when prop is falsy', () => {
    [undefined, null].forEach((falsyProp: undefined| null) => {
      options.propsData.credit = falsyProp;
      const wrapper = shallowMount(DCredit, options);

      const vChip = wrapper.find('v-chip-stub');
      expect(vChip.exists()).toBe(false);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});

// import DCredit from '@/components/course/data/d_credit.vue';

// import { shallowMount } from '@vue/test-utils';
// import { i18n } from '../../../setup';
// import { creditMock } from '../../../data';

// const options = {
//   i18n,
//   propsData: {
//     credit: creditMock,
//   },
// };

// describe('DCredit', () => {
//   it('should not render anything when prop is undefined', () => {
//     options.propsData.credit = undefined;
//     const wrapper = shallowMount(DCredit, options);

//     const span = wrapper.find('span');
//     expect(span.exists()).toBe(false);
//   });

//   it('should not render anything when prop is empty string', () => {
//     options.propsData.credit = null;
//     const wrapper = shallowMount(DCredit, options);

//     const span = wrapper.find('span');
//     expect(span.exists()).toBe(false);
//   });

//   it('should render title data when title prop is correct', () => {
//     const wrapper = shallowMount(DCredit, options);

//     const span = wrapper.find('span');
//     expect(span.text()).toBe(`${creditMock}credit`);
//   });
// });
