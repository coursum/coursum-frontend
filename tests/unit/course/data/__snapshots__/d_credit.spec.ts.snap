// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DCredit should not render anything when prop is falsy 1`] = `<!---->`;

exports[`DCredit should not render anything when prop is falsy 2`] = `<!---->`;

exports[`DCredit should render correct markup when prop is correct 1`] = `
<v-chip-stub
  active="true"
  activeclass=""
  closeicon="$delete"
  filtericon="$complete"
  outlined="true"
  style="color: rgb(146, 146, 146);"
  tag="span"
  xsmall="true"
>
  
  2credit

</v-chip-stub>
`;
